### PR TITLE
Update uncontrolled unit tests

### PR DESCRIPTION
**Pull request by iamwill123**
### Updates to Uncontrolled unit test

- Test button clicks, the image toggles in a modal when clicking on the open and close buttons.
- Test `ESC` key closes modal
- Test mouse scroll closes modal
- Test click away (out of modal, or on modal, or on image) closes the modal

Added helper `user-event` lib for extended user event cases, added `useResizeObserver` jest mock to handle an error todo with `ResizeObserver` being `undefinded` when testing.

### CODE REVIEW

File: source/Uncontrolled.tsx [hunk 1]
```diff
@@ -25,4 +25,4 @@ const Uncontrolled: FC<UncontrolledProps> = (props: UncontrolledProps) => {
 // there is no need to zoom images.
 export default typeof window !== 'undefined'
   ? memo(Uncontrolled)
-  : (props: UncontrolledProps) => props.children
+  : (props: UncontrolledProps) => <Fragment>props.children</Fragment>
```
Comment (rpearce) [inline source/Uncontrolled.tsx:28]: Good call! Did you know about `<></>`?

```suggestion
  : (props: UncontrolledProps) => <>props.children</>
```

File: source/Uncontrolled.test.tsx [hunk 2]
```diff
@@ -1,3 +1,53 @@
-test('<Uncontrolled /> works', () => {
-  expect(true).toEqual(true)
+import { fireEvent, render } from '@testing-library/react'
+import React from 'react'
+import Zoom from '../source'
+
+const ZoomImgTest = () => (
+	<Zoom modalLabelText={`Zoom modal`}>
```
Comment (rpearce) [inline source/Uncontrolled.test.tsx:6]: I need to put a linter in here for this spacing (2 spaces) 😄

**Review by rpearce — APPROVED** [review]
This is superb! Well done, and thank you for your contribution 💜 

imagen

File: source/Uncontrolled.test.tsx (hunk 2 above)
Comment (rpearce) [reply source/Uncontrolled.test.tsx:6]: Oh, I already did. You can see the failures in the build responses below. I recommend installing eslint in your editor and checking out the issues -- you may be able to auto-fix them via your editor, or you can probably run `npm run lint -- --fix`, but I haven't tried that in a hot minute...

File: source/Uncontrolled.tsx (hunk 1 above)
Comment (iamwill123) [reply source/Uncontrolled.tsx:28]: Good morning, yep I know about `<></>` I know it's a shorthand for Fragment, but nothing more than that, I'll update it 👍🏼

File: source/Uncontrolled.test.tsx (hunk 2 above)
Comment (iamwill123) [reply source/Uncontrolled.test.tsx]: ^ fixed, I'll be sure to run the  `lint -- --fix` cmd moving forward.

**Comment by rpearce** [comment]
```
/home/runner/work/image-zoom/image-zoom/stories.js
Error:    3:39  error  'useRef' is defined but never used                   @typescript-eslint/no-unused-vars
Error:   38:7   error  'imgNvidiaCard' is assigned a value but never used   @typescript-eslint/no-unused-vars
Error:   50:7   error  'imgTeAraiPoint' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 3 problems (3 errors, 0 warnings)
```

**Comment by rpearce** [comment]
That's my bad. I'll merge anyway

**Comment by iamwill123** [comment]
sweet! let me know what to test next @rpearce 👍🏼

**Comment by rpearce** [comment]
@iamwill123 Actually, you could do the same tests for the `Controlled` component, but the only difference between the two is that you are now passing `isZoomed` and responding to `onZoomChange` (the component basically saying, "Hey, zoom/unzoom if you want to...seems like a good time to do so").

One additional scenario / train of thought to test would be to re-render the component with the same probs and see if anything changes or not. For example, if you pass `isZoomed={true}` then re-render with a different prop changed (say...`zoomMargin`), is it still zoomed?

**Comment by rpearce** [comment]
Got you in the README on the `main` branch:
imagen

**Comment by iamwill123** [comment]
Awesome @rpearce feels good to contribute 🙏 I will work on the `Controlled` unit test.

**Pull request merged**